crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -58,10 +107,174 @@ impl RequestPump {
                 _ = &mut cancel => {
                     break;
                 }
-                Some(_append) = self.append_stream.next() => {
+                Some(append) = self.append_stream.next() => {
+                    self.handle_append(&provider, append).await;
                 }
             }
         }
+
+        Ok(())
+    }
+
+    /// Infailable handle_append method
+    async fn handle_append<T: TransportConnect>(
+        &mut self,
+        provider: &ReplicatedLogletProvider<T>,
+        incoming: Incoming<Append>,
+    ) {
+        let (reciprocal, append) = incoming.split();
+
+        let loglet = match self.get_loglet(provider, &append.header).await {
+            Ok(loglet) => loglet,
+            Err(err) => {
+                return_error_status!(reciprocal, err);
+            }
+        };
+
+        if !loglet.is_sequencer_local() {
+            return_error_status!(reciprocal, SequencerStatus::NotSequencer);
+        }
+
+        let global_tail = loglet.known_global_tail();
+
+        let loglet_commit = match loglet.enqueue_batch(append.payloads).await {
+            Ok(loglet_commit) => loglet_commit,
+            Err(err) => {
+                return_error_status!(reciprocal, SequencerStatus::from(err), global_tail);
+            }
+        };
+
+        let task = WaitAppendedTask {
+            reciprocal,
+            loglet_commit,
+            global_tail: global_tail.clone(),
+        };
+
+        let _ = task_center().spawn_child(TaskKind::Disposable, "wait-appended", None, task.run());
+    }
+
+    async fn get_loglet<T: TransportConnect>(
+        &self,
+        provider: &ReplicatedLogletProvider<T>,
+        header: &CommonRequestHeader,
+    ) -> Result<Arc<ReplicatedLoglet<T>>, SequencerStatus> {
+        let loglet = match provider.get_active_loglet(header.log_id, header.segment_index) {
+            Some(loglet) => loglet,
```

Comment:
It would be nice if you validate that this loglet has the loglet_id that we expect from the append header. Just as a sanity check.